### ISSUE TITLE
RFC Removing top level rules

### ISSUE THREAD

**Issue by TimWhiting** [UserNobody14/tree-sitter-dart#43]
I'm planning on removing the top level rules for statements and expressions.

They were present for tests, but I've migrated tests to not need them anymore. I think we might be able to get rid of some of the rule conflicts if we clean up the grammar a bit, and this is a first stab at doing so. I'm working on adding records and patterns, but they will take some time (just finished record literals today), and the grammar just keeps getting more complex. This is an attempt at reducing the complexity so this implementation is easier to maintain.

Let me know if you foresee any issues with removing these rules. I'm going to preemptively remove them so that I can make progress on the dart 3.0 features, but I can revert any of my recent changes if it breaks downstream users.
@theHamsta @akinsho @UserNobody14 

Add anyone else you feel needs to see this.
In general I think the biggest user of the grammar is neovim. I don't see why any users would be using top-level expressions or statements and expect them to parse. Only top-level definitions should parse.

I think if anyone would like non-standard start rules we should redirect them to this issue: https://github.com/tree-sitter/tree-sitter/issues/711

**Comment by theHamsta** [comment]
@amaanq

**Comment by TimWhiting** [comment]
Closing, for now. Dart 3.0 features are implemented and are ready for more broad testing by downstream users, though you might want to have a rollback plan in case of failure since there were so many changes. I did test on the whole flutter repo, and only get an error on a line that actually is not good syntax, so I'm not too worried. Let me know if you find issues, and I'll try to be prompt at responding.
